openpaas-james/deployment-tests/common/src/main/java/com/linagora/openpaas/deployment/CliContract.java

```diff
@@ -0,0 +1,18 @@
+package com.linagora.openpaas.deployment;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.GenericContainer;
+
+public interface CliContract {
+    GenericContainer<?> jamesContainer();
+
+    @Test
+    default void cliShouldWork() throws Exception {
+        jamesContainer().execInContainer("james-cli", "addDomain", "domain.tld");
+
+        assertThat(jamesContainer().execInContainer("james-cli", "ListDomains").getStdout())
```

Comment:
I'm not sure testing only stdout gives me enough confidence in this test. Shouldn't we test we really added the domain?

Reply:
how? by checking if webadmin works, which should maybe be an other test?

Reply:
I would, **please**, like to keep the deployment test counts **as low as possible** as they are really long to play.

I am unwilling to have more than a single test for the CLI (which by the way is a net gain!)

Maybe that means breaking other testing rules like single assertion.

Let me suggest:

```
    default void cliShouldWork() throws Exception {
        assertThat(jamesContainer().execInContainer("james-cli", "ListDomains").getStdout())
            .doesNotContain("domain.tld");
            
        jamesContainer().execInContainer("james-cli", "addDomain", "domain.tld");

        assertThat(jamesContainer().execInContainer("james-cli", "ListDomains").getStdout())
```

Doing asserts on return codes is an option too.